pkg/waiter/manager.go

```diff
@@ -0,0 +1,255 @@
+package waiter
+
+import (
+	"context"
+	"errors"
+
+	"github.com/stackrox/rox/pkg/sync"
+)
+
+const (
+	// The default number of attempts to re-generate an ID on collision.
+	defaultMaxCollisions = 5
+)
+
+var (
+	// ErrTooManyCollisions the ID generator produced too many non-unique IDs.
+	ErrTooManyCollisions = errors.New("too many id collisions")
+
+	// ErrManagerShutdown manager has been shutdown and therefore cannot process requests.
+	ErrManagerShutdown = errors.New("manager is shutdown")
+
+	errKeyExists = errors.New("id exists")
+	errNoExists  = errors.New("id does not exist")
+)
+
+type response[T any] struct {
+	id   string
+	data T
+	err  error
+}
+
+type options struct {
+	idGenerator   IDGenerator
```

Comment:
in case you want to abstract the UUID generation, just create a unexported type alias here (assuming you are doing this to make it testable).

Reply:
I removed this based on Connors comments above. It was both for testing and not, I had used this structure in a personal project where different generators were needed depending on scenario - it worked well so I carried it over. Now it is no longer an option to change the generator.